libs/network/include/network/generics/requesting_queue.hpp

```diff
@@ -85,10 +85,12 @@ class RequestingQueueOf
   Counters Resolve();
   Counters Resolve(Timepoint const &time_point);
 
-  bool HasCompletedPromises() const;
-  bool HasFailedPromises() const;
-  void DiscardFailures();
-  void DiscardCompleted();
+  bool     HasCompletedPromises() const;
+  bool     HasFailedPromises() const;
+  void     DiscardFailures();
+  void     DiscardCompleted();
+  bool     Empty() const;
```

Comment:
Added empty and size functionality for this queue so as to determine whether we are hammering the other node(s) with requests. 

This class should probably be looked at in time, it looks like failed connetions will never be dropped based on the way we are using it.